Core/Generators/CPlusPlus/CPlusPlusGenerator.cs

```diff
@@ -237,7 +237,7 @@ private string CompileDecodeField(TypeBase type, string target, int depth = 0, i
                     $"{tab}{target}{dot}reserve(length{depth});" + nl +
                     $"{tab}for (size_t {i} = 0; {i} < length{depth}; {i}++) {{" + nl +
                     $"{tab}{tab}{TypeName(at.MemberType)} x{depth};" + nl +
-                    $"{tab}{tab}{CompileDecodeField(at.MemberType, $"x{depth}", depth + 1, indentDepth + 2, isOptional)}" + nl +
+                    $"{tab}{tab}{CompileDecodeField(at.MemberType, $"x{depth}", depth + 1, indentDepth + 2, false)}" + nl +
```

Comment:
I forgot to mention this, but this is a bugfix.

This boolean parameter essentially tracks whether the target is actually wrapped in `std::optional`. But when a message contains an array field, we are not dealing with `optional<vector<optional<MemberType>>>`: The elements themselves are not optional, so this "inner" call should not propagate `isOptional`.